lstchain/reco/dl1_to_dl2.py

```diff
@@ -374,40 +375,55 @@ def build_models(filegammas, fileprotons, features,
     df_gamma = pd.read_hdf(filegammas, key='events/LSTCam')
     df_proton = pd.read_hdf(fileprotons, key='events/LSTCam')
 
-    df_gamma = filter_events(df_gamma, r_max=r_max, intensity_min=intensity_min)
-    df_proton = filter_events(df_proton, r_max=r_max, intensity_min=intensity_min)
+    df_gamma = filter_events(df_gamma,
+                             leakage_cut=leakage_cut,
+                             intensity_min=intensity_min,
+                             r_min=r_min)
+    df_proton = filter_events(df_proton,
+                              leakage_cut=leakage_cut,
+                              intensity_min=intensity_min,
+                              r_min=r_min)
 
     #Train regressors for energy and disp_norm reconstruction, only with gammas
 
-    reg_energy, reg_disp = train_reco(df_gamma, features,
-                                      regression_args=regression_args
-                                      )
+    reg_energy = train_energy(df_gamma, features,
+                            regression_args=regression_args
+                          )
+    reg_disp_vector = train_disp_vector(df_gamma, features,
+                            regression_args=regression_args
+                          )
 
     #Train classifier for gamma/hadron separation.
 
     train, testg = train_test_split(df_gamma, test_size=0.2)
     test = testg.append(df_proton, ignore_index=True)
 
-    tempRFreg_Energy, tempRFreg_Disp = train_reco(train, features_,
-                                                  regression_args=regression_args,
-                                                  )
+    temp_reg_energy = train_energy(train, features,
```

Comment:
This is also something I meant to discuss in the past.
Why training an (or two) extra temporary regressor and not using the one(s) trained before?

Reply:
I decided to do this way in order to have the Energy and Disp regressors trained with the higher possible statistics, without separating events in train/test. Afterwards, I divide the sample in train and test to train the smaller regressors and reconstruct energy and disp for the gamma/hadron separation. I thought this way the regressors for energy and disp stored to use in other samples (completely different from the ones used for training) would be better. 
Maybe it's not the best way to proceed¿? but I thought that this way we could have the best regressors, and make sure that we were not mixing samples when training/testing g/h separation.  Maybe now that we have a much bigger MC production this is not necessary anyway, what do you think?

Reply:
Hi.
I see. But then, how do you assess the energy/direction reconstruction performance? you will need a (different) validation dataset anyway...

Reply:
Yes, and indeed I have  a separate dataset for testing energy and direction reco. Either I test on point like gammas or in a separate set of diffuse gammas.